### DISCUSSION TITLE
Comments on the emitterReceiverSchemeTutorial

### DISCUSSION THREAD

**Discussion by tsampazk** [aidudezzz/deepbots-tutorials#15] [General]
[Link to the tutorial.](https://github.com/aidudezzz/deepbots-tutorials/tree/master/emitterReceiverSchemeTutorial)

**Comment by startgram** [comment]
Hi! Thanks a lot for you work! I'm starting only this topic and i have some errors in my console:
DEPRECATION: Robot.getEmitter is deprecated, please use Robot.getDevice instead.
DEPRECATION: Robot.getReceiver is deprecated, please use Robot.getDevice instead.
Traceback (most recent call last):
  File "supervisorController.py", line 93, in <module>
    supervisor = CartPoleSupervisor()
  File "supervisorController.py", line 14, in __init__
    self.respawnRobot()
  File "supervisorController.py", line 32, in respawnRobot
    childrenField.importMFNode(-2, "CartPoleRobot.wbo")	 # Load robot from file and add to second-to-last position
AttributeError: 'Field' object has no attribute 'importMFNode'
WARNING: 'supervisorController' controller exited with status: 1.

**Reply by tsampazk** [reply]
Hello there! Sorry for the errors, the tutorial is currently a little bit out of date. There have been lots of changes in Webots recently. 

If i may ask, do you have any specific reason to use the emitter-receiver scheme instead of the [robot-supervisor](https://github.com/aidudezzz/deepbots-tutorials/tree/master/robotSupervisorSchemeTutorial)? It should work without issues on the latest version and it is much faster and simpler!

Let me know what version of Webots and deepbots you are using and how you installed deepbots (regularly through pip or from test.pypi?).

**Comment by MehdiShahbazi** [comment]
Thank you for the tutorial. It’s incredibly insightful, and I want to express my gratitude for the effort you and your team have invested in creating it.

**Reply by tsampazk** [reply]
Thank you for your kind words @MehdiShahbazi ! Happy to help. :smile:
